File: SDKs/Aspose.Cells_Cloud_SDK_For_.NET/src/Com/Aspose/cells/Model/Column.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Com.Aspose.Cells.Model {
  public class Column {
    public int? GroupLevel { get; set; }

    public int? Index { get; set; }

    public bool? IsHidden { get; set; }

    public double? Width { get; set; }

    public LinkElement Style { get; set; }

    public Link link { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class Column {\n");
      sb.Append("  GroupLevel: ").Append(GroupLevel).Append("\n");
      sb.Append("  Index: ").Append(Index).Append("\n");
      sb.Append("  IsHidden: ").Append(IsHidden).Append("\n");
      sb.Append("  Width: ").Append(Width).Append("\n");
      sb.Append("  Style: ").Append(Style).Append("\n");
      sb.Append("  link: ").Append(link).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }

File: SDKs/Aspose.Cells-Cloud-SDK-for-Ruby/lib/aspose_cells_cloud/models/column.rb
module AsposeCellsCloud
  # 
  class Column < BaseObject
    attr_accessor :group_level, :index, :is_hidden, :width, :style, :link
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'group_level' => :'GroupLevel',
        
        # 
        :'index' => :'Index',
        
        # 
        :'is_hidden' => :'IsHidden',
        
        # 
        :'width' => :'Width',
        
        # 
        :'style' => :'Style',
        
        # 
        :'link' => :'link'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'group_level' => :'Integer',
        :'index' => :'Integer',
        :'is_hidden' => :'BOOLEAN',
        :'width' => :'Float',
        :'style' => :'LinkElement',
        :'link' => :'Link'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'GroupLevel']
        self.group_level = attributes[:'GroupLevel']
      end
      
      if attributes[:'Index']
        self.index = attributes[:'Index']
      end
      
      if attributes[:'IsHidden']
        self.is_hidden = attributes[:'IsHidden']
      end
      
      if attributes[:'Width']
        self.width = attributes[:'Width']
      end
      
      if attributes[:'Style']
        self.style = attributes[:'Style']
      end
      
      if attributes[:'link']
        self.link = attributes[:'link']
      end
      
    end

  end
end
